packages/module/src/CullingInfo/CullingInfo.tsx

```diff
@@ -0,0 +1,134 @@
+export interface CullingInformation extends Omit<TooltipProps, 'content'> {
+  /** Option to add custom css classes */
+  className?: string;
+  /** Warning date for when object becomes stale */
+  staleWarning: CullingDate;
+  /** Date when object becomes culled */
+  culled: CullingDate;
+  /** Date when object becomes stale */
+  stale: CullingDate;
+  /** Current date */
+  currDate: CullingDate;
+  /** Optional prop to add custom children */
+  children?: React.ReactElement<any, string | React.JSXElementConstructor<any>> | undefined;
+  /** Option to add custom message ReactElement */
+  render?: Render;
+  /** Optional custom warning message */
+  message?: string;
+}
+
+const CullingInformation: React.FunctionComponent<CullingInformation> = ({
+  culled = new Date(0),
+  className,
+  staleWarning = new Date(0),
+  stale = new Date(0),
+  currDate = new Date(0),
+  children,
+  render,
+  message,
+  ...props
+}) => {
+  const classes = useStyles();
+
+  const calculateTooltip: CalculateTooltip = (culled, warning, currDate) => {
+    const culledDate: Date = new Date(culled);
+    const warningDate: Date = new Date(warning);
+    const diffTime: number = new Date(currDate).valueOf() - warningDate.valueOf();
+    const removeIn: number = Math.ceil((culledDate.valueOf() - new Date(currDate).valueOf()) / days);
+    const msg = message ? message : `System scheduled for inventory removal in ${removeIn} days`;
+    if (diffTime >= 0) {
+      return {
+        isError: true,
+        msg,
+      };
+    }
+  
+    return {
+      isWarn: true,
+      msg,
+    };
+  };
+
+  if (new Date(currDate).valueOf() - new Date(stale).valueOf() < 0) {
+    return render
+      ? render({
+        msg: '',
+      })
+      : children || null;
+  }
+
+  const { isWarn, isError, msg }: CullingInfo = calculateTooltip(culled, staleWarning, currDate);
+  if (render) {
+    return (
+      <span
+        className={clsx({ [classes.inventoryCullingWarning]: isWarn, [classes.inventoryCullingDanger]: isError }, className)}
+      >
+        {isWarn &&
+          <Button variant="plain" aria-label="Action" role="tooltip" icon={<ExclamationTriangleIcon className={clsx( classes.iconMargin, classes.inventoryCullingWarning )}/>} />   
+        }
+        {isError &&
+          <Button variant="plain" aria-label="Action" role="tooltip" icon={<ExclamationCircleIcon  className={clsx( classes.iconMargin, classes.inventoryCullingDanger )}/>} />   
+        }
+        <span className={clsx( classes.messageFont )}>
+          {render({ msg })}
+        </span>
+        
+      </span>
+    );
+  }
+
+  return (
+    <React.Fragment>
+      <Tooltip {...props} content={msg} position="bottom">
+        <span
+          className={clsx({ [classes.inventoryCullingWarning]: isWarn, [classes.inventoryCullingDanger]: isError }, className)}
+        >
+          {isError && <ExclamationCircleIcon />}
+          {isWarn && <ExclamationTriangleIcon />}
+          {children}
```

Comment:
The first example is still not screen reader accessible - and it may be true that the button needs to replace the span - or the Tooltip needs to be placed on the button within the span. 

you can see that the first example is screen reader inaccessible in the video below. you can also see how the tooltip is not being triggered by the screenreader since I think the tooltip needs to be placed on the button itself rather than the span outside the button.

i recommend testing the examples by tabbing through them as a sanity check - if they are keyboard accessible, it's much more likely that it'll be screen reader accessible.

![Jan-30-2025 17-27-38](https://github.com/user-attachments/assets/081fde21-92e1-4953-b624-44785effb662)

Reply:
oops I forgot to add the buttons to the tooltip version of the CullingInfo. I was able to tab through them after the fix.